library/openwrt_setup.sh

```diff
@@ -38,6 +38,11 @@ main() {
     json_add_string ansible_distribution_release "$dist_release"
     json_add_string ansible_distribution_version "$dist_version"
     json_add_string ansible_os_family OpenWRT
+    if [ "$(ls -di / | awk '{print $1}')" != "$(ls -di /proc/1/root/. | awk '{print $1}')" ]; then
+        json_add_string ansible_is_chroot "true"
+    else
+        json_add_string ansible_is_chroot "false"
+    fi
```

Comment:
This should work:
```suggestion
    json_add_boolean ansible_is_chroot "$(test -r /proc/1/root/. &&
        { test / -ef /proc/1/root/.; echo $?; } ||
        { test "$(ls -di / | cut -d' ' -f1)" -eq 2; echo $?; }
    )"
```
Although I've taken the `2` from the Ansible implementation. Could you check what `ls -di /` gives on an actual OpenWRT device please? I've only tested this in Docker, which is some kind of chroot.

Without the fallback, this should do:
```suggestion
    json_add_boolean ansible_is_chroot "$(test / -ef /proc/1/root/.; echo $?)"
```

Reply:
Thank you for your thoughts on this.
1. Yes, a boolean makes a lot of sense.
2. Using the `-ef` test makes a lot of sense as well. It's not POSIX, but seems to work on OpenWRT's BusyBox.
3. Using `cut` to get the inode number does not work for me. `ls` indents the inode number with several spaces, so it's not field 1 for `cut`. That's why I used `awk`. I'm open to a nicer solution.
4. The matter of the fallback is somewhat difficult.
  I have an x86_64 VM and a TL-WDR 4300 running fresh installations of OpenWRT 22.03. `/` has inode `2` on both devices.
  I also have another TL-WDR 4300 running OpenWRT 19.07, which has been upgraded from several older version. Unlike the others, this latter device uses extroot and the inode of `/` is `281`.
  Also, when I boot my OpenWRT VM using a live Linux system and chroot to the disk with OpenWRT, the inode of `/` in the chroot is (not surprisingly) `2`.
  So, the fallback check may be wrong. But it's (probably) the same for Ansible's original `setup`.